repos/fdbt-site/src/pages/api/managePurchaseMethod.ts

```diff
@@ -86,12 +93,59 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
                     id: 'purchase-method-name',
                 });
             }
+
+            if (id) {
+                const purchaseDetails = await getSalesOfferPackageByIdAndNoc(id, noc);
+                if (purchaseDetails.isCapped) {
+                    const dbIsCapped = purchaseDetails.isCapped;
+
+                    const validPurchaseLocations = purchaseLocationsList(dbIsCapped).method;
+                    const isValidPurchaseLocation = toArray(purchaseLocations).every((val) =>
+                        validPurchaseLocations.includes(val),
+                    );
+
+                    if (!isValidPurchaseLocation) {
+                        errors.push({
+                            errorMessage: `Select the valid option(s) for the purchase locations.`,
+                            id: purchaseLocationsList(dbIsCapped).id,
+                        });
+                    }
+
+                    const validPaymentMethods = paymentMethodsList(dbIsCapped).paymentMethods;
+                    const isValidPaymentMethod = toArray(paymentMethods).every((val) =>
+                        validPaymentMethods.includes(val),
+                    );
+
+                    if (!isValidPaymentMethod) {
+                        errors.push({
+                            errorMessage: `Select the valid option(s) for the payment method.`,
+                            id: paymentMethodsList(dbIsCapped).id,
+                        });
+                    }
+
+                    const validTicketFormats = ticketFormatsList(dbIsCapped).ticketFormats.map(
+                        (ticketFormat) => ticketFormat.value,
+                    );
+                    const isValidTicketFormat = toArray(ticketFormats).every((val) => validTicketFormats.includes(val));
+
+                    if (!isValidTicketFormat) {
+                        errors.push({
+                            errorMessage: `Select the valid option(s) for the ticket format.`,
+                            id: ticketFormatsList(dbIsCapped).id,
+                        });
+                    }
+                }
+            }
         }
 
         if (errors.length > 0) {
             updateSessionAttribute(req, GS_PURCHASE_METHOD_ATTRIBUTE, { inputs: salesOfferPackage, errors });
-
-            redirectTo(res, `/managePurchaseMethod${id ? `?id=${id}` : ''}`);
+            redirectTo(
+                res,
+                `/managePurchaseMethod${
+                    id ? (isCapped ? `?isCapped=true&id=${id}` : `?id=${id}`) : isCapped ? '?isCapped=true' : ''
```

Comment:
Can we split this out so its not all inline? Its a very difficult line to read.

Instead, we just need something like 

`const cappedPart = isCapped ? 'isCapped=true' : ''; `
and
`const idPart = id ? 'id=${id}' : '';`

and then
`const andSymbol = !!cappedPart && !!idPart ? '&' : '';`

`const redirectString = '/managePurchaseMethod${cappedPart}${andSymbol}${idPart}';`

Its more code, but its clearer on how the redirect part is being built up